code/core/system/proxy-velodyne64/testsuites/ProxyVelodyne64TestSuite.h

```diff
@@ -22,21 +22,202 @@
+                        if (velodyneFrame.getComponentDataType() == SharedPointCloud::FLOAT_T && (velodyneFrame.getNumberOfComponentsPerPoint() == 4) && (velodyneFrame.getUserInfo() == SharedPointCloud::XYZ_INTENSITY)) {
+                            memcpy(m_velodyneMemory->getSharedMemory(), vsm->getSharedMemory(), velodyneFrame.getSize());
+                            cout << "Copy decoded data in Frame 1 to a memory segment" << endl;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    mutable uint8_t m_counter;
+    mutable std::shared_ptr< odcore::wrapper::SharedMemory > m_velodyneMemory;
+};
+
+class packetToByte : public odcore::io::conference::ContainerListener {
+   public:
+    packetToByte(std::shared_ptr< odcore::wrapper::SharedMemory > m1, std::shared_ptr< odcore::wrapper::SharedMemory > m2)
+        : mcc(m2)
+        , v64decoder(m1, mcc, "../db.xml") //The calibration file db.xml is automatically copied to the parent folder of the test suite binary
+    {}
+
+    ~packetToByte() {}
+
+    virtual void nextContainer(odcore::data::Container &c) {
+        if (c.getDataType() == odcore::data::pcap::GlobalHeader::ID()) {
+            cout << "Get the global header" << endl;
+        }
+        if (c.getDataType() == odcore::data::pcap::PacketHeader::ID()) {
+            //cout<<"Received a packet!"<<endl;
+        }
+        if (c.getDataType() == odcore::data::pcap::Packet::ID()) {
+            // Here, we have a valid packet.
+
+            //Extract payload from the packet
+            pcap::Packet packet = c.getData< pcap::Packet >();
+            pcap::PacketHeader packetHeader = packet.getHeader();
+            if (packetHeader.getIncl_len() == 1248) {
+                string payload = packet.getPayload();
+                payload = payload.substr(42, 1206); //Remove the 42-byte Ethernet header
+                v64decoder.nextString(payload);     //Let the Velodyne64 decoder decode the 1206-byte payload
+            }
+        }
+    }
+
+   private:
+    MyContainerConference mcc;
+    opendlv::core::system::proxy::velodyne64Decoder v64decoder;
+};
 
 class ProxyVelodyne64Test : public CxxTest::TestSuite {
    public:
-    void setUp() {}
+    ProxyVelodyne64Test()
+        : velodyneSM(SharedMemoryFactory::createSharedMemory(NAME, SIZE))
+        , segment(SharedMemoryFactory::createSharedMemory(NAME2, SIZE)) {}
+
+    ~ProxyVelodyne64Test() {}
+
+    void readCsvFile() {                                              //Read a .csv file that is compressed into a .zip file
+        fstream fin("../atwallFrame1New.zip", ios::binary | ios::in); //Read xyz and intensity values of points in Frame 1 exported from VeloView. These values exist in a zip file in the parent folder
+        if (!fin.is_open()){
```

Comment:
As we are in a test case, you can make the test case fail if the file is not present. Just replace the if(...) {..} with TS_ASSERT(fin.is_open());